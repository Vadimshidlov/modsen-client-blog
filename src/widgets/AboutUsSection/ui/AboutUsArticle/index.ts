export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
