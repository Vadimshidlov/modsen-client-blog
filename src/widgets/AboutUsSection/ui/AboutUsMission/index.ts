export { AboutUsMission } from '@/widgets/AboutUsSection/ui/AboutUsMission/AboutUsMission';
