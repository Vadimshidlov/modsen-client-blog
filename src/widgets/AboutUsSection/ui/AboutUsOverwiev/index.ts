export { AboutUsOverwiev } from '@/widgets/AboutUsSection/ui/AboutUsOverwiev/AboutUsOverwiev';
