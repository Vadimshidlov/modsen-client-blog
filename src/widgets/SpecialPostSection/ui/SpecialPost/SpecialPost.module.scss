.contentBlock {
  position: absolute;
  z-index: 1;
  width: 50%;
  right: 0;
  bottom: 0;
  display: flex;
  flex-direction: column;
  padding: 121px 80px 80px;
  background: $white;

  @media (max-width: 1200px) {
    position: static;
    width: 100%;

    padding: 80px 60px 60px;
  }

  @media (max-width: 768px) {
    padding: 40px 10px 0;
  }

  .postSubTitle {
    font-size: $fz24;
    margin-bottom: 24px;

    @media (max-width: 768px) {
      font-size: $fz20;
      margin-bottom: 12px;
    }
  }

  .postTitle {
    font-size: $fz48;
    font-weight: $bold;
    line-height: 64px;
    margin-bottom: 16px;

    @media (max-width: 768px) {
      font-size: $fz32;
      line-height: 50px;
      margin-bottom: 8px;
    }

    @media (max-width: 450px) {
      font-size: $fz28;
    }
  }

  .postText {
    max-width: 546px;
    font-size: $fz16;
    font-weight: $regular;
    line-height: 28px;
    margin-bottom: 32px;
  }
}
