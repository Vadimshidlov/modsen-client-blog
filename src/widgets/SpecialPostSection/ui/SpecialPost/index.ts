export { SpecialPost } from '@/widgets/SpecialPostSection/ui/SpecialPost/SpecialPost';
