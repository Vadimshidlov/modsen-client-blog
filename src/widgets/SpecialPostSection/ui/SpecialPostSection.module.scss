.specialPostSection {
  position: relative;
  display: flex;
  margin-bottom: 128px;

  @media (max-width: 1200px) {
    flex-direction: column;
    align-items: center;
  }

  @media (max-width: 768px) {
    margin-bottom: 64px;
  }

  @media (max-width: 450px) {
    margin-bottom: 32px;
  }

  .postImage {
    width: 70%;
    object-fit: cover;
    height: auto;

    @media (max-width: 1200px) {
      width: 100%;
      height: 50%;
    }
  }
}
