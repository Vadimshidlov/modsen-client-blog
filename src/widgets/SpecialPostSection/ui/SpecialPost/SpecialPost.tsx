import React from 'react';
import styles from '@/widgets/SpecialPostSection/ui/SpecialPost/SpecialPost.module.scss';
import { Inter } from 'next/font/google';
import { Button } from '@/shared/ui/Button';
import Link from 'next/link';

const inter = Inter({
  subsets: ['latin'],
  weight: '700',
  variable: '--font-inter',
});

export function SpecialPost() {
  return (
    <article className={styles.contentBlock}>
      <h3 className={`${styles.postSubTitle} ${inter.variable}`}>WHY WE STARTED</h3>
      <h2 className={styles.postTitle}>
        It started out as a simple idea and evolved into our passion
      </h2>
      <span className={styles.postText}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
        labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
        laboris nisi ut aliquip.
      </span>
      <Button>
        <Link href="/about">Discover our story &gt;</Link>
      </Button>
    </article>
  );
}
