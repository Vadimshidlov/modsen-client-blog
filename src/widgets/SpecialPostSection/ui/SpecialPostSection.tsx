import React from 'react';
import Image from 'next/image';
import { SpecialPost } from '@/widgets/SpecialPostSection/ui/SpecialPost';
import specialPostImageSrc from '@/widgets/SpecialPostSection/assets/close-up-photography-of-man-wearing-sunglasses-1212984.jpg';
import container from '@/shared/styles/container.module.scss';
import styles from '@/widgets/SpecialPostSection/ui/SpecialPostSection.module.scss';

export function SpecialPostSection() {
  return (
    <section className={`${styles.specialPostSection} ${container.container}`}>
      <Image
        src={specialPostImageSrc}
        className={styles.postImage}
        placeholder="blur"
        alt="special-post-image"
      />
      <SpecialPost />
    </section>
  );
}
