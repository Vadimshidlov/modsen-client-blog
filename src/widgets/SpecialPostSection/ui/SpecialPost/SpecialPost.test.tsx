import React from 'react';
import { render } from '@testing-library/react';
import { SpecialPost } from '@/widgets/SpecialPostSection/ui/SpecialPost';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('SpecialPost tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<SpecialPost />);
    expect(container).toMatchSnapshot();
  });
});
