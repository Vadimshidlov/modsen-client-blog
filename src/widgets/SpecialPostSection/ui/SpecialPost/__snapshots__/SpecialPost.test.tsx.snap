// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecialPost tests should matches snapshot 1`] = `
<div>
  <article
    class="contentBlock"
  >
    <h3
      class="postSubTitle variable"
    >
      Mocked title
    </h3>
    <h2
      class="postTitle"
    >
      Mocked postTitle
    </h2>
    <span
      class="postText"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
    </span>
    <button
      class="button"
      type="button"
    >
      <a
        href="/about"
      >
        Mocked buttonText
         &gt;
      </a>
    </button>
  </article>
</div>
`;
